### PR TITLE
Add metrics reporting

### PR DESCRIPTION
**Pull request by syndicut**
I added metrics reporting to support monitoring remco's state. This PR is inspired by this https://github.com/hashicorp/consul-template/pull/1378 consul-template PR and metrics support in consul itself. Metrics are reported by go-metrics library (which is also used in consul and planned to be used in consul-template: https://github.com/hashicorp/consul-template/issues/1395).

Exposed metrics:
  - **files.template_execution_duration**
    - Duration of template execution
  - **files.check_command_duration**
    - Duration of check_command execution
  - **files.reload_command_duration**
    - Duration of reload_command execution
  - **files.stage_errors_total**
    - Total number of errors in file staging action
  - **files.staged_total**
    - Total number of successfully files staged
  - **files.sync_errors_total**
    - Total number of errors in file syncing action
  - **files.synced_total**
    - Total number of successfully files synced
  - **backends.sync_errors_total**
    - Total errors in backend sync action
  - **backends.synced_total**
    - Total number of successfully synced backends

### CODE REVIEW

File: pkg/telemetry/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	prometheusSink *metricsPrometheus.PrometheusSink
+}
+
+func (p *PrometheusSink) Init() (metrics.MetricSink, error) {
+	if p == nil {
+		return nil, ErrNilConfig
+	}
+
+	handler := http.NewServeMux()
+	handler.Handle("/metrics", promhttp.Handler())
```
Comment (HeavyHorst) [inline pkg/telemetry/prometheus.go:30]: Could you add a hint to the docs that the metrics are available under addr/metrics? 
I just tried addr and had to look up the handler path in the code.

File: pkg/telemetry/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	p.httpServer = &http.Server{Addr: p.Addr, Handler: handler}
+
+	go func() {
+		err := p.httpServer.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			log.Error(fmt.Sprintf("error starting prometheus server: %v", err))
```
Comment (HeavyHorst) [inline pkg/telemetry/prometheus.go:36]: I think we only start the prometheus stats endpoint and not a real prometheus server.
Could you please clarify this in this log message?

**Review by HeavyHorst — COMMENTED** [review]
This is great!

Just 2 nitpicks.

LGTM otherwise.

**Comment by syndicut** [comment]
Updated PR with fixes to our comments and other small fixups. Hope you would not mind adding me to authors list as mentioned in contributing guide.

**Comment by HeavyHorst** [comment]
Thanks, great addition.

LGTM

**Pull request merged**